lib/proxy.js

```diff
@@ -94,9 +94,9 @@ function buildStream (host, options, secure) {
 ////////////////////////////////////////////////////////////
 
 function buildResultingMiddleware (host, options, stream) {
-  console.log(stream)
   return function proxyMiddleware (req, res, next) {
 
+    var _stream = _.clone(stream, true);
```

Comment:
i am concerned about the cost of a clone, but a shallow clone of an array of function references isnt the end of the world. This shouldnt cause the problem since line 99 is inside the `proxyMiddleware` function which is called in request, not on server startup https://github.com/johnhof/express-proxy/blob/master/lib/proxy.js#L98